karapace/schema_registry.py

```diff
@@ -33,9 +34,9 @@
 X_REGISTRY_VERSION_HEADER = ("X-Registry-Version", f"karapace-{__version__}".encode("utf8"))
 
 
-def _resolve_version(subject_data: SubjectData, version: Version) -> ResolvedVersion:
-    max_version = max(subject_data["schemas"])
-    resolved_version: int
+def _resolve_version(schema_versions: Dict[ResolvedVersion, SchemaVersion], version: Version) -> ResolvedVersion:
```

Comment:
suggestion (non-blocking): Consider using `Mapping` in function arguments instead of `Dict`, as it helps in distinguishing where mutation happens.